code/datums/components/mood.dm

```diff
@@ -28,6 +28,11 @@
 		modify_hud()
 		var/datum/hud/hud = owner.hud_used
 		hud.show_hud(hud.hud_version)
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	if(H.mind?.assigned_role in list("Research Director", "Scientist", "Roboticist"))
```

Comment:
Gene should be included since they are part of science somewhat

Reply:
done